ipfs/mock.go

```diff
@@ -1,22 +1,51 @@
 package ipfs
 
 import (
-	"io"
+	"context"
 
+	nilrouting "github.com/ipfs/go-ipfs-routing/none"
+	"github.com/ipfs/go-ipfs/core"
+	coremock "github.com/ipfs/go-ipfs/core/mock"
 	ipld "github.com/ipfs/go-ipld-format"
-	mdutils "github.com/ipfs/go-merkledag/test"
 	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	"github.com/libp2p/go-libp2p-core/routing"
+	mocknet "github.com/libp2p/go-libp2p/p2p/net/mock"
+
+	"github.com/lazyledger/lazyledger-core/ipfs/plugin"
 )
 
 // Mock provides simple mock IPFS API useful for testing
 func Mock() APIProvider {
-	return func() (coreiface.APIDagService, io.Closer, error) {
-		dom := dagOnlyMock{mdutils.Mock()}
+	return func() (coreiface.APIDagService, *core.IpfsNode, error) {
+		plugin.EnableNMT()
+
+		nd, err := MockNode()
+		if err != nil {
+			return nil, nil, err
+		}
 
-		return dom, dom, nil
+		return dagOnlyMock{nd.DAG}, nd, nil
 	}
```

Comment:
> Should we also keep a more minimal version like we did before? 
For now, I don't see any reason to.

Reply:
Removing `Close` is a nice catch actually 👍🏻

Reply:
> if not, then we can delete the Close method for dagOnlyMock

Yeah, also I just realized that returning DagService instead of APIDagService would be a good change. The difference between them is just in Pinning API that we won't use anyway and in return we fully remove coreiface package dependency.

Furthermore, DAGService is a component of IpfsNode, thus we can get it straightly from it, so I guess APIProvider can be already morphed into NodeProvider already and that is good IMO, but that's not part of the PR.

Reply:
Sounds good, I think I can fit it into #374, but if not I can write a quick separate PR. We might run into some of the CI issues we did before when mocking a Node, but I think we can be clever enough to get around it should that occur.